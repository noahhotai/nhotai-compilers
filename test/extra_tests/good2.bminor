/* string format expr */

x: integer = 65;
s: string = "hello";
b: boolean = false;
d: array [2] array [2] float = {{1.2, 2.1}, {3.1, 3.5}};

min: function integer ( a: integer, b: integer ) =
{
	if(a > b) return a;
	else return b;
}

testfunc: function void ( z: integer ) =
{
	print "minimum", min(10, 5), "\n", z=10, z==10;
}


