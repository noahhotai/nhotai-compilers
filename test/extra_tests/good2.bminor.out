Program parsed successfully.
a resolves to param 1
b resolves to param 2
a resolves to param 1
b resolves to param 2
min resolves to global min
z resolves to param 1
z resolves to param 1
Program resolved successfully.
Program type checked successfully.
